mobile/src/shared/components/Layout.tsx

```diff
@@ -1,12 +1,82 @@
-import type { FunctionComponent, ReactElement } from "react";
-import { Outlet } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { Link as RouterLink, Outlet } from "react-router-dom";
+
+import {
+  AppBar,
+  Box,
+  Container,
+  Link,
+  MenuItem,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+
+import type { User } from "../../user/UserTypes";
+import type { UserRepository } from "../../user/UserRepository";
+import { NullableUser } from "../SharedTypes";
+
+interface LayoutProps {
+  userRepository: UserRepository;
+}
+
+const Layout = ({ userRepository }: LayoutProps) => {
```

Comment:
Why do you need to pass the repository at all ? Can't you just make the `useUserRepository` return one whenever it is needed?

Reply:
We need to pass the repository here since useUserRepository relies on the Outlet context (see https://reactrouter.com/en/main/hooks/use-outlet-context). The Layout is in charge of setting this context.

Reply:
It could just not rely on the outlet context and instantiate the repository itself right?